zingolib/src/wallet/keys/unified.rs

```diff
@@ -352,6 +353,19 @@ impl WalletCapability {
         Ok(Self::new_from_seed(config, &bip39_seed, position))
     }
 
+    /// Creates a new `WalletCapability` from a unified spending key.
+    pub fn new_from_usk(usk: &[u8]) -> Result<Self, String> {
+        // Decode unified spending key
+        let usk = UnifiedSpendingKey::from_bytes(Era::Orchard, usk)
+            .map_err(|_| "Error decoding unified spending key.")?;
+        Ok(Self {
+            orchard: Capability::Spend(usk.orchard().to_owned()),
+            sapling: Capability::Spend(usk.sapling().to_owned()),
+            // transparent: Capability::Spend(usk.transparent()),
```

Comment:
Does anyone know how to translate an `legacy::AccountPrivKey` to an `ExtendedPrivKey`?

The private key that we have here is the `m/44'/133'/account'` derivation. I'm not certain whether `WalletCapability` wants that, or a `m/44'/133'/account'/0/0` derivation, but I suspect the former, so that it can read 'change' UTXOs.

Reply:
`AccountPrivKey` apparently has exactly one field, which is an `ExtendedPrivKey`. So no conversion necessary. Just access to that field, which is _not_ public, is required. Ironically, `zcash_primitives` doesn't appear to expose that field for reading at all(!). So I may be forced to call its `to_bytes()` method to serialize it out, and then deserialize it. What a pain.

Reply:
https://github.com/zcash/librustzcash/issues/929